avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1

```diff
@@ -954,6 +907,184 @@ Describe 'Module tests' -Tag 'Module' {
       $templateFileContent.metadata.owner | Should -Not -BeNullOrEmpty
     }
   }
+
+  Context 'User-defined-types tests' -Tag 'UDT' {
+
+    $udtTestCases = [System.Collections.ArrayList] @() # General UDT tests (e.g. param should exist)
+    $udtSpecificTestCases = [System.Collections.ArrayList] @() # Specific UDT test cases for singular UDTs (e.g. tags)
+    foreach ($moduleFolderPath in $moduleFolderPaths) {
+
+      $resourceTypeIdentifier = ($moduleFolderPath -split '[\/|\\]{1}avm[\/|\\]{1}(res|ptn)[\/|\\]{1}')[2] -replace '\\', '/' # avm/res/<provider>/<resourceType>
+
+      # For runtime purposes, we cache the compiled template in a hashtable that uses a formatted relative module path as a key
+      $moduleFolderPathKey = $moduleFolderPath.Replace('\', '/').Split('/avm/')[1].Trim('/').Replace('/', '-')
+      if (-not ($convertedTemplates.Keys -contains $moduleFolderPathKey)) {
+        if (Test-Path (Join-Path $moduleFolderPath 'main.bicep')) {
+          $templateFilePath = Join-Path $moduleFolderPath 'main.bicep'
+          $templateContent = bicep build $templateFilePath --stdout | ConvertFrom-Json -AsHashtable
+
+          if (-not $templateContent) {
+            throw ($bicepTemplateCompilationFailedException -f $templateFilePath)
+          }
+        } else {
+          throw ($templateNotFoundException -f $moduleFolderPath)
+        }
+        $convertedTemplates[$moduleFolderPathKey] = @{
+          templateContent  = $templateContent
+          templateFilePath = $templateFilePath
+        }
+      } else {
+        $templateContent = $convertedTemplates[$moduleFolderPathKey].templateContent
+        $templateFilePath = $convertedTemplates[$moduleFolderPathKey].templateFilePath
+      }
+
+      $udtSpecificTestCases += @{
+        moduleFolderName         = $resourceTypeIdentifier
+        templateFileContent      = $templateContent
+        templateFileContentBicep = Get-Content $templateFilePath
+      }
+
+      # Setting expected URL only for those that doen't have multiple different variants
+      $avmInterfaceSpecsTemplateBase = 'https://raw.githubusercontent.com/Azure/Azure-Verified-Modules/main/docs/static/includes/interfaces'
+      $avmInterfaceSpecsBase = 'https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces'
```

Comment:
I'd suggest to create vanity links to not be dependent on specific links

Reply:
Ok. @matebarabas could you do that? I guess you should own the link rather than @eriqua or me.

Reply:
@eriqua, vanity URLs created as per internal discussion. CC: @AlexanderSehr

Reply:
@eriqua, updated and re-tested with KVLT